core/src/main/scala/org/scalacheck/ops/time/JavaInstantGenerators.scala

```diff
@@ -1,43 +1,48 @@
 package org.scalacheck.ops.time
 
-import java.time.temporal._
-import java.time.{Duration, Instant}
+import java.time.{Clock, Duration, Instant}
+
+import org.scalacheck.{Arbitrary, Gen}
 
 object JavaInstantGenerators extends JavaInstantGenerators
 trait JavaInstantGenerators extends AbstractTimeGenerators {
   override type InstantType = Instant
-  override type DurationType = TemporalAmount
-  override type ParamsType = JavaTimeParams
+  override type DurationType = Duration
+  override type ParamsType = Clock
 
-  override protected[time] def datetime(millis: Long)
-    (implicit params: JavaTimeParams): Instant = Instant.ofEpochMilli(millis)
+  override protected[time] def now(implicit clock: Clock): Instant = Instant.now(clock)
 
-  override protected[time] def duration(millis: Long): TemporalAmount = Duration.ofMillis(millis)
+  override val defaultRange: Duration = Duration.ofDays(365)
 
-  override protected[time] def millis(duration: TemporalAmount): Long = duration.get(ChronoUnit.NANOS) / 1000
+  override val defaultParams: Clock = Clock.systemUTC()
 
-  override protected[time] def millis(datetime: Instant)
-    (implicit params: JavaTimeParams): Long = datetime.toEpochMilli
+  override def between(start: Instant, end: Instant)(implicit params: Clock): Gen[Instant] = {
+    for {
+      seconds <- Gen.choose(start.getEpochSecond, end.getEpochSecond)
+      nanos <- {
+        if (seconds == end.getEpochSecond) Gen.choose(start.getNano, end.getNano)
```

Comment:
`start.getNano` can be greater than `end.getNano`.
If `start.getEpochSecond == end.getEpochSecond` this could be used (because we assume start < end), otherwise `Gen.choose(0, end.getNano)`

Reply:
Good catch!